crates/node/src/commands/run.rs

```diff
@@ -52,6 +52,10 @@ pub struct ExtendedRunCmd {
     /// increases the memory footprint of the node.
     #[clap(long)]
     pub cache: bool,
+
+    /// When enable, the node will sync state from l1,
+    #[clap(long)]
+    pub sync_from_l1: Option<String>,
```

Comment:
In `ExtendedRunCommand` `pub base: RunCmd, `pub network_params: NetworkParams,` ` pub sync: SyncMode,` we find this, provided by substrate:
```rust
/// Syncing mode.
#[derive(Debug, Clone, Copy, ValueEnum, PartialEq)]
#[value(rename_all = "kebab-case")]
pub enum SyncMode {
	/// Full sync. Download end verify all blocks.
	Full,
	/// Download blocks without executing them. Download latest state with proofs.
	Fast,
	/// Download blocks without executing them. Download latest state without proofs.
	FastUnsafe,
	/// Prove finality and download the latest state.
	Warp,
}
```
Do you think we can reuse those already defined modes? They look a bit similar to what we offer.

What is the list of the different sync mode we aim to support? I think we should define them.

Reply:
Has this made it into the codebase in the end?

Reply:
The `SyncMode` defined in substrate specifies the synchronization method between nodes within the same network. However, "sync from l1" is another type of synchronization that only exists within the rollup chain. Therefore, we cannot reuse the Substrate SyncMode. Additionally, extending Substrate `SyncMode` is not quite suitable since substrate does not provide Rollup functionality. Therefore, it is more appropriate to continue placing this command parameter in the `ExtendedRunCommand`.

Reply:
>https://github.com/Generative-Labs/madara/blob/state-sync-from-l1/crates/node/src/commands/run.rs#L113